cmd/infra/ibmcloud_powervs/create.go

```diff
@@ -396,109 +356,87 @@ func (infra *Infra) createCloudInstance(option *CreateInfraOptions) (resourceIns
 		return nil, fmt.Errorf("create cloud instance returned nil")
 	}
 
-	if *resourceInstance.State == "active" {
+	if *resourceInstance.State == activeStateL {
 		return resourceInstance, nil
 	}
 
-	retryExceeds := false
-	ttl := make(chan bool, 1)
-	go func() {
-		for {
-			if retryExceeds {
-				break
-			}
-			resourceInstance, _, err = rcv2.GetResourceInstance(&resourcecontrollerv2.GetResourceInstanceOptions{ID: resourceInstance.ID})
-			log.Log.Info("Waiting for cloud instance to up", "id", resourceInstance.ID, "state", *resourceInstance.State)
-			if *resourceInstance.State == "active" || err != nil {
-				break
-			}
-			time.Sleep(time.Second)
+	f := func() (bool, error) {
+		resourceInstance, _, err = rcv2.GetResourceInstance(&resourcecontrollerv2.GetResourceInstanceOptions{ID: resourceInstance.ID})
+		log.Log.Info("Waiting for cloud instance to up", "id", resourceInstance.ID, "state", *resourceInstance.State)
+
+		if err != nil {
+			return false, err
 		}
-		ttl <- true
-	}()
 
-	select {
-	case <-time.After(10 * time.Minute):
-		retryExceeds = true
-		return nil, fmt.Errorf("time exceeds waiting for cloud instance to up")
-	case <-ttl:
+		if *resourceInstance.State == activeStateL {
+			return true, nil
+		}
+		return false, nil
 	}
 
+	err = wait.PollImmediate(time.Second*5, time.Minute*5, f)
+
 	return resourceInstance, err
 }
 
-func createPowerVSSession(pvRegion string) (session *ibmpisession.IBMPISession, err error) {
-	auth := getIAMAuth()
-	account, err := servicesUtils.GetAccount(auth)
-
+func getAccount(auth core.Authenticator) (string, error) {
```

Comment:
I somehow feel this is a hack atm becacause we are parsing from the bearer toke, lets add a TODO to replace this logic with proper IBM Cloud API to get an account ID for any given API key

Reply:
I have copied the capi code directly here, need to explore whether any API available to get the account id since we are getting this before creating PowerVS session, PowerVS api won't be usable.

Reply:
yes, lets add the TODO and explore later..

Reply:
there is an API available in IAM identitytv1 API, implemented using that, please take a look